Client/src/Components/AttributeFilter/AttributeFilterUtils.ts

```diff
@@ -152,6 +152,26 @@ export function getTree(ontologyEntries: Iterable<Field>): FieldTreeNode {
     : { field: GENERATED_ROOT, children: rootChildren };
 }
 
+export function getLeavesOfSubTree(ontologyEntries: Iterable<Field>, rootTerm: Field): Field[] {
+  const entriesByParentTerm = mapBy(ontologyEntries, term => term.parent) as Map<string, Field[]>;
+  let result = [] as Field[];
+  let nodesToSearch = [rootTerm] as Field[];
+  while(nodesToSearch.length > 0 ){
+    let nodesToSearchNext = [] as Field[];
+    for(const node of nodesToSearch){
+      const children = entriesByParentTerm.get(node.term);
+      if(children != null && children.length > 0){
+        nodesToSearchNext = nodesToSearchNext.concat(children);
+      } else {
+        result.push(node);
+      }
```

Comment:
This logic isn't quite right. We want to push any node that can be used as a filter. There are cases where this can be a branch node. See this utility function:  https://github.com/VEuPathDB/WDKClient/blob/master/Client/src/Components/AttributeFilter/AttributeFilterUtils.ts#L30-L32

Reply:
Thank you for finding this, I replaced it with isFilterField!